File: code/java/searchMatrixII.java
class Solution {
    public boolean searchMatrixII(int[][] matrix, int target) {
        if(matrix.length == 0)
            return false;
        if(matrix[0].length == 0)
            return false;
        int len = matrix[0].length;
        for(int i=0;i<matrix.length;i++){
            if(target >=matrix[i][0] && target <= matrix[i][len-1]){
                for(int j=0;j<len;j++)
                    if(matrix[i][j] == target)
                        return true;
                    else if(matrix[i][j] > target)
                        break;
            }
            if(target < matrix[i][0])
                return false;
        
        }       
        return false;
    }
}

File: code/python/searchMatrixII.py
class Solution:
    def searchMatrixII(self, matrix, target):
        """
        :type matrix: List[List[int]]
        :type target: int
        :rtype: bool
        """
        if len(matrix) == 0:
            return False
        if len(matrix[0])  == 0:
            return False
        for i in range(len(matrix)):
            if(target >= matrix[i][0] and target <= matrix[i][-1]):
                if target in matrix[i]:
                    return True
            if target < matrix[i][0]:
                return False
        return False
